src/mono/wasm/Wasm.Build.Tests/Blazor/BlazorWasmTestBase.cs

```diff
@@ -98,15 +98,26 @@ public string CreateBlazorWasmTemplateProject(string id)
         bool publish = false,
         bool setWasmDevel = true,
         params string[] extraArgs)
-        => BuildProjectWithoutAssert(
-                    id,
-                    config,
-                    new BuildProjectOptions(CreateProject: false, UseCache: false, Publish: publish),
-                    extraArgs.Concat(new[]
-                    {
-                        "-p:BlazorEnableCompression=false",
-                        setWasmDevel ? "-p:_WasmDevel=true" : string.Empty
-                    }).ToArray());
+    {
+        try
+        {
+            return BuildProjectWithoutAssert(
+                        id,
+                        config,
+                        new BuildProjectOptions(CreateProject: false, UseCache: false, Publish: publish),
+                        extraArgs.Concat(new[]
+                        {
+                            "-p:BlazorEnableCompression=false",
+                            setWasmDevel ? "-p:_WasmDevel=true" : string.Empty
+                        }).ToArray());
+        }
+        catch (XunitException xe)
+        {
+            if (xe.Message.Contains("error CS1001: Identifier expected"))
```

Comment:
Could we do some regex to capture all C# compiler errors?

Reply:
Do we need that? This is for a very specific failure, see https://github.com/dotnet/runtime/issues/86533 .

Reply:
In this case we generate a project with `dotnet new blazorwasm` but then it fails to compile. But that doesn't happen for *all* the times we do that in the tests, just sometimes with some random test. So, I want to find out what that Program.cs has. I suspect it might be some error message, or output of something.

Reply:
guess: do we need to flush some file before we start compilation process ?

Reply:
Nope. This is a exec of "dotnet new" and that should return only once the write is "effectively" done. And the failure is during a subsequent build.

Reply:
compiler or msbuild service sticking around?

Reply:
`dotnet new blazorwasm` followed by regex hacking the `Program.cs` ?

Reply:
> compiler or msbuild service sticking around?

I doubt it. But will know more in the next few days/weeks when this is hit again.

Reply:
> `dotnet new blazorwasm` followed by regex hacking the `Program.cs` ?

Hm I hadn't considered this. I will keep this in mind. 👍